src/mongocrypt-key.c

```diff
@@ -101,9 +104,11 @@ _parse_masterkey (bson_iter_t *iter,
             out->masterkey_provider = MONGOCRYPT_KMS_PROVIDER_AWS;
          } else if (0 == strcmp (provider, "local")) {
             out->masterkey_provider = MONGOCRYPT_KMS_PROVIDER_LOCAL;
+         } else if (0 == strcmp (provider, "azure")) {
+            out->masterkey_provider = MONGOCRYPT_KMS_PROVIDER_AZURE;
          } else {
             CLIENT_ERR (
-               "invalid 'masterKey.provider', expected 'aws' or 'local'");
+               "invalid 'masterKey.provider', expected 'aws' or 'local' or 'azure' or 'gcp'");
```

Comment:
It is not uncommon for tests to check error messages for substrings (libmongocrypt has its own tests like so). I left this with slightly odd wording to prevent unnecessary friction.